src/compiler/parser.ts

```diff
@@ -6259,7 +6259,7 @@ namespace ts {
                 }
 
                 function parseJSDocIdentifier(): Identifier {
-                    if (token !== SyntaxKind.Identifier) {
+                    if (token !== SyntaxKind.Identifier && !isTSOnlyKeyword(token)) {

```

Comment:
We don't want to allow *all* keywords though. If the keyword is a JS keyword, it should be a syntax error.